core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java

```diff
@@ -83,8 +84,8 @@ protected synchronized String issueCrumb(ServletRequest request, String salt) {
                     buffer.append(req.getSession().getId());
                 }
 
-                md.update(buffer.toString().getBytes());
-                return Util.toHexString(md.digest(salt.getBytes()));
+                md.update(buffer.toString().getBytes(Charset.defaultCharset()));
+                return Util.toHexString(md.digest(salt.getBytes(Charset.defaultCharset())));
```

Comment:
ASCII or UTF-8

Reply:
The buffer can contain an authentication principal, so let's use UTF-8 for that. The salt is a hexadecimal string, so ASCII should suffice.